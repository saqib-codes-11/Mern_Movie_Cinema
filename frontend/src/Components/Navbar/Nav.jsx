import { useState, useEffect } from "react";
import { Navbar, Nav, NavDropdown } from 'react-bootstrap';
import Container from 'react-bootstrap/Container';
import { Form, Button } from 'react-bootstrap';
import Modal from 'react-bootstrap/Modal'
import axios from 'axios'
import Logo from '../../Images/Logo.png';
import { Link } from 'react-router-dom';
import './NavBar.css';

const NavBar = () => {
    const [show, setShow] = useState(false);
    const [searchQuery, setSearchQuery] = useState("");
    const [movieList, setMovieList] = useState([]);
    const handleClose = () => setShow(false);
    const handleShow = () => setShow(true);

    const getSearchResults = () => {
        axios.get(`http://localhost:5000/movie/search?s=${searchQuery}`)
            .then((response) => {
                setMovieList(response.data);
            })
    }

    useEffect(() => {
        getSearchResults();
    }, [searchQuery])

    const releasedChecker = (released, title, id) => {
        if (released) {
            return (
                <Link to={`/released/${id}`} onClick={handleClose} style={{ color: "white" }}>
                    <h5>{title}</h5>
                </Link>
            )
        } else if (!released) {
            return (
                <Link to={`/upcoming/${id}`} onClick={handleClose} style={{ color: "white" }}>
                    <h5>{title}</h5>
                </Link>
            )
        }
    }

    return (
        <Navbar className="NavBar" variant="dark" expand="lg">
            <Container>
                <Navbar.Brand href="/Home">
                    <img
                        alt=""
                        src={Logo}
                        width="125"
                        height="35"
                        className="d-inline-block align-top"
                    /></Navbar.Brand>
                <Navbar.Toggle aria-controls="basic-navbar-nav" />
                <Navbar.Collapse id="basic-navbar-nav">
                    <Nav className="mr-auto">
                        <Nav.Link href="/Home">Home</Nav.Link>
                        <NavDropdown title="Movies" id="basic-nav-dropdown">
                            <NavDropdown.Item href="/CurrentMovies">Current Movies</NavDropdown.Item>
                            <NavDropdown.Item href="/UpcomingMovies">Upcoming Movies</NavDropdown.Item>
                        </NavDropdown>
                        <Nav.Link href="/Tickets">Tickets</Nav.Link>
                        <Nav.Link href="/Screens">Screens</Nav.Link>
                        <Nav.Link href="/OpeningTimes">Opening Times</Nav.Link>
                        <Nav.Link className="nav-links" href="/Forum">Forum</Nav.Link>
                        <Nav.Link href="/Classifications/0">Classifications</Nav.Link>
                    </Nav>
                    <Form inline>
                        <Button variant="outline-danger" onClick={handleShow} block>Search</Button>
                        <Modal show={show} onHide={handleClose} className="bgBlur searchModal">
                            <Modal.Header className="searchModalHeader" closeButton>
                                <Modal.Title>Search for something!</Modal.Title>
                            </Modal.Header>
                            <Modal.Body>
                                <Form.Control type="text" placeholder="Search Term" value={searchQuery} onChange={(e) => setSearchQuery(e.target.value)} />
                                <div className="pt-2">
                                    {movieList.map((movie) => (
                                        releasedChecker(movie.released, movie.title, movie._id)
                                    ))}
                                </div>
                            </Modal.Body>
                            <Modal.Footer>
                                <Button variant="outline-light" onClick={handleClose}>
                                    Close
                                </Button>
                            </Modal.Footer>
                        </Modal>
                    </Form>
                </Navbar.Collapse>
            </Container >
        </Navbar >
    );
}
export default NavBar;