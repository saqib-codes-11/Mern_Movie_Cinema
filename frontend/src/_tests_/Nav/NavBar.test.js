import React from 'react';
import Nav from '../../Components/Navbar/Nav';
import renderer from 'react-test-renderer';
import { BrowserRouter } from "react-router-dom";

describe(`testing the navigation`, () => {

    let movieArr = [
        {
            "_id": "60e429f57a5d6042284b2faf",
            "title": "The Black Phone",
            "released": false
        },
        {
            "_id": "60e41e337a5d6042284b2f4c",
            "title": "Black Widow",
            "released": true
        }
    ];
    let creator;
    let testComponent;

    // test("It matches the snapshot", () => {
    //     const tree = renderer.create(<Nav />).toJSON();
    //     expect(tree).toMatchSnapshot();

    // })

    it(`tests the search function`, () => {
        const realUseState = React.useState

        jest.spyOn(React, 'useState')
            .mockImplementationOnce(() => realUseState(false))
            .mockImplementationOnce(() => realUseState("black"))
            .mockImplementationOnce(() => realUseState(movieArr))

        const tree = renderer.create(<BrowserRouter><Nav /></BrowserRouter>).toJSON();
        expect(tree).toMatchSnapshot();


    })

})

