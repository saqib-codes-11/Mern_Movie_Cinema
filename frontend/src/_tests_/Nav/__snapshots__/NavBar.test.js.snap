// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the navigation tests the search function 1`] = `
<nav
  className="NavBar navbar navbar-expand-lg navbar-dark"
>
  <div
    className="container"
  >
    <a
      className="navbar-brand"
      href="/Home"
    >
      <img
        alt=""
        className="d-inline-block align-top"
        height="35"
        src="Logo.png"
        width="125"
      />
    </a>
    <button
      aria-controls="basic-navbar-nav"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
      id="basic-navbar-nav"
    >
      <div
        className="mr-auto navbar-nav"
        onKeyDown={[Function]}
      >
        <a
          className="nav-link"
          data-rb-event-key="/Home"
          href="/Home"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          Home
        </a>
        <div
          className="nav-item dropdown"
        >
          <a
            aria-expanded={false}
            aria-haspopup={true}
            className="dropdown-toggle nav-link"
            data-rb-event-key={null}
            href="#"
            id="basic-nav-dropdown"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
          >
            Movies
          </a>
        </div>
        <a
          className="nav-link"
          data-rb-event-key="/Tickets"
          href="/Tickets"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          Tickets
        </a>
        <a
          className="nav-link"
          data-rb-event-key="/Screens"
          href="/Screens"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          Screens
        </a>
        <a
          className="nav-link"
          data-rb-event-key="/OpeningTimes"
          href="/OpeningTimes"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          Opening Times
        </a>
        <a
          className="nav-links nav-link"
          data-rb-event-key="/Forum"
          href="/Forum"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          Forum
        </a>
        <a
          className="nav-link"
          data-rb-event-key="/Classifications/0"
          href="/Classifications/0"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          Classifications
        </a>
      </div>
      <form
        className="form-inline"
      >
        <button
          className="btn btn-outline-danger btn-block"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Search
        </button>
      </form>
    </div>
  </div>
</nav>
`;
